### PR TITLE
feat: update path mapping page and make it read only for APIs created by gko

### PR DESCRIPTION
**Pull request by kamiiiel**
## Issue

https://gravitee.atlassian.net/browse/APIM-434

## Description

Migrate old path mapping to the new Angular version and make the path mappings read-only if they were created using Kubernetes operator

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-zwdhxdbmzk.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    limitations under the License.
+
+-->
+<h2 mat-dialog-title class="title">Add Path Mapping</h2>
+
+<mat-dialog-content class="content">
+  <mat-tab-group dynamicHeight mat-stretch-tabs (selectedTabChange)="onSelectedTab($event)">
+    <!-- Add Path Mapping -->
+    <mat-tab [label]="tabLabels.AddPathMapping">
+      <div class="content__url-tab">
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html]: Here both tabs have the same css and we doesn't really have any url in this dialog so I would rename the class for something like "content__tab" ( here and on the line 48 )

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    width: 100%;
+    padding: 0;
+  }
+
+  &__url-tab {
+    padding-top: 16px;
+
+    &__field {
+      width: 100%;
+    }
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.scss]: I think this class is not used we can remove it

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        <mat-dialog-content class="content" [formGroup]="pathFormGroup">
+          <mat-form-field class="content__path__form-field" appearance="fill">
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html]: <ma-error ...> are cropped.

Screenshot 2023-01-20 at 14 56 04

I think we can avoid it by replacing the `padding: 0` by a   `padding-bottom: 4px;`

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+          </mat-form-field>
+        </mat-dialog-content>
+      </div>
+    </mat-tab>
+
+    <!-- Import Swagger -->
+    <mat-tab [label]="tabLabels.SwaggerDocument">
+      <div class="content__url-tab">
+        <mat-dialog-content class="content" [formGroup]="pathFormGroup">
+          <mat-radio-group formControlName="path" class="radio-group" data-testid="designer-template-parameter-case">
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html]: The first time I select a radio button I have the error telling  me that the path mapping doesn't respect the correct pattern is it the wanted behaviour ?

Screenshot 2023-01-20 at 15 05 06

Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html]: Maybe we should align it in column and not in line no wdyt ?
```
.radio-group {
  display: flex;
  flex-direction: column;

  &__button {
    margin: 8px 0 0 0;
  }
}
```

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings.component.ts [hunk 5]
```diff
@@ -45,25 +51,36 @@ export class ApiPathMappingsComponent implements OnInit, OnDestroy {
...
       .get(this.ajsStateParams.apiId)
       .pipe(
-        tap((api) => {
+        mergeMap((api) => {
           this.api = api;
           this.pathMappingsDS = this.toPathMappingDS(api);
           this.isLoadingData = false;
+
+          this.isReadOnly = api.definition_context.origin === ('kubernetes' as ApiOrigin);
+          return this.documentationService.apiSearch(api.id, {
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings.component.ts]: Maybe we can do it independently to the `apiservice.get` as we have already access to the api id using this.ajsStateParams.apiId. Wdyt ?

**Review by Okhelifi — COMMENTED** [review]
Good job. Some small comments here

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html (hunk 1 above)
Comment (kamiiiel) [reply gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html]: I updated the code

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.scss (hunk 2 above)
Comment (kamiiiel) [reply gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.scss]: Updated

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html (hunk 3 above)
Comment (kamiiiel) [reply gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html:26]: This is something that I wanted to talk about it. Do we want to have the same checks for both pages or not? The old page doesn't check

Comment (kamiiiel) [reply gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html]: done

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html (hunk 4 above)
Comment (kamiiiel) [reply gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html]: I removed the path validation

Comment (kamiiiel) [reply gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-add-dialog/api-path-mappings-add-dialog.component.html]: done

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings.component.ts (hunk 5 above)
Comment (kamiiiel) [reply gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings.component.ts]: updated

**Review by a-cordier — APPROVED** [review]

**Pull request closed without merging**